packages/admin-next/dashboard/src/routes/discounts/details/discount-detail.tsx

```diff
@@ -0,0 +1,65 @@
+import { useAdminDiscount } from "medusa-react"
+import { Outlet, useLoaderData, useParams } from "react-router-dom"
+
+import { JsonViewSection } from "../../../components/common/json-view-section"
+import { DiscountGeneralSection } from "./components/discounts-general-section"
+import { DiscountConfigurationSection } from "./components/discounts-configurations-section"
+
+import { discountLoader } from "./loader"
+import { RedemptionsSection } from "./components/redemptions-section"
+import { DetailsSection } from "./components/details-section"
+import { DiscountConditionsSection } from "./components/discounts-conditions-section"
+
+import before from "medusa-admin:widgets/discount/details/before"
+import after from "medusa-admin:widgets/discount/details/after"
+
+export const DiscountDetail = () => {
+  const initialData = useLoaderData() as Awaited<
+    ReturnType<typeof discountLoader>
+  >
+
+  const { id } = useParams()
+  const { discount, isLoading } = useAdminDiscount(id!, {
+    initialData,
+  })
```

Comment:
Currently the initial data is being passed to the query params

```suggestion
  const { discount, isLoading } = useAdminDiscount(id!, undefined, {
    initialData,
  })
```